### PR TITLE
Filter() docs: Clarify 'collect telemetry' vs 'instrumentation'

### PR DESCRIPTION
**Pull request by kevinmckinley**
## Changes

Clarify `Filter()` identifies requests that are desired for collecting telemetry (vs instrumentation)
reference: https://github.com/open-telemetry/opentelemetry-dotnet/issues/1186

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #1186
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.AspNet/AspNetInstrumentationOptions.cs [hunk 1]
```diff
@@ -26,7 +26,7 @@ namespace OpenTelemetry.Instrumentation.AspNet
     public class AspNetInstrumentationOptions
     {
         /// <summary>
-        /// Gets or sets a Filter function to filter instrumentation for requests on a per request basis.
+        /// Gets or sets a Filter function to collect telemetry about requests on a per request basis.
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.AspNet/AspNetInstrumentationOptions.cs:29]:
```suggestion
        /// Gets or sets a Filter function which can decide to collect or not collect telemetry about requests on a per request basis.
```

Comment (cijothomas) [reply src/OpenTelemetry.Instrumentation.AspNet/AspNetInstrumentationOptions.cs:29]: Is this better?

Comment (kevinmckinley) [reply src/OpenTelemetry.Instrumentation.AspNet/AspNetInstrumentationOptions.cs:29]: `Gets or sets a Filter function that determines whether or not to collect telemetry about requests on a per request basis`?

I'm fine with either. You tell me and I'll update it 👍

Comment (cijothomas) [reply src/OpenTelemetry.Instrumentation.AspNet/AspNetInstrumentationOptions.cs:29]: Gets or sets a Filter function that determines whether or not to collect telemetry about requests on a per request basis -- this looks good for me.

Comment (kevinmckinley) [reply src/OpenTelemetry.Instrumentation.AspNet/AspNetInstrumentationOptions.cs]: Updated.
* AspNet
* AspNetCore
* Http

**Review by cijothomas — APPROVED** [review]

**Pull request merged**